src/node/nodetonode.h

```diff
@@ -89,23 +117,44 @@ namespace ccf
       return t;
     }
 
+    template <class T>
+    std::vector<uint8_t> recv_authenticated_with_load(
+      const uint8_t*& data, size_t& size)
+    {
+      // data contains the message header of type T, the raw data, and the gcm
+      // header at the end
+      const auto& t = serialized::peek<T>(data, size);
+      const auto d = serialized::read(data, size, (size - sizeof(GcmHdr)));
+      const auto& hdr = serialized::overlay<GcmHdr>(data, size);
+
+      auto& n2n_channel = channels->get(t.from_node);
+
+      if (!n2n_channel.verify(hdr, d))
+      {
+        throw std::logic_error(fmt::format(
```

Comment:
Please do not crash here, we should not kill the process here

Reply:
The process is not killed, this is caught in pbft.h